server/src/main/java/com/objectcomputing/checkins/services/memberprofile/currentuser/CurrentUserServicesImpl.java

```diff
@@ -63,6 +76,35 @@ public MemberProfile getCurrentUser() {
         throw new MemberProfileDoesNotExistException("No active members in the system");
     }
 
+    @Override
+    public boolean isCurrentUserPdlFor(UUID memberId) {
+        if (!hasRole(RoleType.PDL)) {
+            return false;
+        }
+        MemberProfile currentUser = getCurrentUser();
+        MemberProfile checkUser = memberProfileRepo.findById(memberId).orElseThrow(() -> {
+            throw new NotFoundException("No such member exists");
+        });
+
+        if (checkUser.getPdlId().equals(currentUser.getId())) {
+            return true;
+        }
+
+        List<Team> memberTeams = teamRepository.matchMemberAndlead(memberId.toString(), currentUser.getId().toString());
```

Comment:
Per my understanding, this piece of code is mentioning that if teamMember and TeamLead (Current user) are part of the same team, and if teamLead is the currentUser then return true (that currentUser is PDL for memberID). Is this correct?

could you elaborate on why this validation is required?